### PR TITLE
Refactor/plan invocation

### PR DESCRIPTION
**Pull request by wederbn**
#### Short Description
Refactors the plan invocation process by moving the instance handling for plan instances to the management bus.

#### Proposed Changes

  * moves plan instance creation to the management bus
  * moves updating of plan outputs to the management bus
  * removes some legacy code related to correlation handling
  * transforms the PlanInvocationEngine to a proxy that is only in charge of tranforming the parameters to an OSGi Event and sending it to the bus (could be removed completely in a next step)

### CODE REVIEW

File: org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import org.apache.camel.Message;
 import org.opentosca.bus.management.deployment.plugin.IManagementBusDeploymentPluginService;
 import org.opentosca.bus.management.header.MBHeader;
-import org.opentosca.bus.management.service.impl.collaboration.Constants;
+import org.opentosca.bus.management.service.impl.Constants;
```
Comment (nyuuyn) [inline org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java]: Was this essential to the change? But LGTM :)

File: org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/instance/plan/PlanInstanceHandler.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            LOG.debug("Instance ID of the plan in Camunda: {}", planInstanceID);
+
+            // create web resource to retrieve the current state of the process instance
+            final Client client = Client.create();
+            WebResource webResource =
+                Client.create()
+                      .resource(Settings.ENGINE_PLAN_BPMN_URL + Constants.PROCESS_INSTANCE_PATH + planInstanceID);
+
+            // wait until the process instance terminates
+            while (true) {
```
Comment (nyuuyn) [inline org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/instance/plan/PlanInstanceHandler.java]: Isn't this blocking some other things? The method is static as well 🤔

File: org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java [hunk 3]
```diff
@@ -1,326 +1,49 @@
 package org.opentosca.container.core.impl.plan;
```
Comment (nyuuyn) [inline org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java]: Wouldn't it be better to remove the whole when it only does this much? :)

File: org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java (hunk 1 above)
Comment (wederbn) [reply org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java]: By moving the plan instance handling to the bus, we need some constants not related to the collaboration between different containers. Therefore, we either need two constants classes or move the existing one from the collaboration to the general package and I think thats the better solution :-)

File: org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java (hunk 3 above)
Comment (wederbn) [reply org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java]: In principle yes. The logic for creating the OSGi event could be moved to the calling classes (API, situations). However, we need the possibility to create a correlation ID somewhere in the core. So we could move this method somewhere else, but the question is where? :-)

Comment (Vogel612) [reply org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java]: Considering that OSGI is no longer used in the winery-model branch, I don't think it's useful to duplicate the work for that

File: org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java (hunk 1 above)
Comment (Vogel612) [reply org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java]: I personally dislike the idea of a `Constants` class for various reasons, the primary one being that names should be scoped to the smallest applicable scope. There is precedent for multiple `Constants` classes, though. Examples include `MBHeaders`, vast majorities of the Settings, which are functionally constant, as well as classes used for Routing configuration within the main bus.

**Pull request merged**